File: LA_7.2/SimpleStack.java
import java.io.*;
public class SimpleStack {
    public static void main (String [] args) {
        Scanner sc=new Scanner (System.in);
        final int max=5;
        String [] st =new String [max];
        GenSimpleStack&lt;String&gt; stack=new GenSimpleStack&lt;String&gt;(st);
        while (true) {
            int ch=sc.nextInt();
            switch(ch) {
                        case 0:
                            System.exit(1);
                        case 1:
                            try {
                                 stack.push(sc.next());
                            }
                            catch(StackFullException e) {
                                 System.out.print(e);
                            }
                            break;
                        case 2: 
                            try {
                                stack.pop();
                            }
                            catch (StackEmptyException e) {
                                System.out.print(e);
                            }
                            break;
                        }
          }
    }
}                         
interface IGenSimpleStack&lt;T&gt; {
    public void push (T item) throws StackFullException ;
    public void pop() throws StackEmptyException;
    public boolean isFull();
    public boolean isEmpty();
}
    class GenSimpleStack&lt;T&gt;  {
        T [] data;
        int tos;
        GenSimpleStack(T [] ds) {
            this.data=ds;
            tos=0;
        }
        public void push (T item) throws StackFullException {
            if (isFull()) 
                throw new StackFullException(data.length);
            data[tos]=item;
            tos++;
            
        }
        public void pop() throws StackEmptyException {
            if (isEmpty()) 
                throw new StackEmptyException();
            tos--;
            System.out.print(data[tos] + " ");
          }
          public boolean isFull() {
              return tos==data.length;
          }
          public boolean isEmpty() {
              return tos==0;
          
       }
 } 
    class StackFullException extends Exception {
    int size;
    StackFullException(int s) {
        super("StackFull");
        size=s;
    }
    public String toString() {
        return ("[Full-5] ");
    }
}
    class StackEmptyException extends Exception {
        StackEmptyException() {
            super("StackEmpty");
        }
        public String toString() {
            return ("[Empty]");
        }
    }

File: LA_7.2/SimpleStack.cpp
#include <iostream>
#include <stdexcept>
#include <string>
using namespace std;
class StackFullException : public exception {
public:
    int size;
    StackFullException(int s) : size(s) {}
    const char* what() const noexcept override {
        return "[Full-5] ";
    }
};
class StackEmptyException : public exception {
public:
    const char* what() const noexcept override {
        return "[Empty]";
    }
};
template <typename T>
class GenSimpleStack {
    T* data;
    int tos;
    int maxSize;
public:
    GenSimpleStack(T* ds, int max) : data(ds), maxSize(max), tos(0) {}
    void push(T item) {
        if (isFull()) {
            throw StackFullException(maxSize);
        }
        data[tos++] = item;
    }
    void pop() {
        if (isEmpty()) {
            throw StackEmptyException();
        }
        cout << data[--tos] << " ";
    }
    bool isFull() const {
        return tos == maxSize;
    }
    bool isEmpty() const {
        return tos == 0;
    }
};
int main() {
    const int max = 5;
    string st[max];
    GenSimpleStack<string> stack(st, max);
    while (true) {
        int ch;
        cout << "Enter 0 to exit, 1 to push, 2 to pop: ";
        cin >> ch;

        switch (ch) {
            case 0:
                exit(0);
            case 1: {
                string item;
                cout << "Enter item to push: ";
                cin >> item;
                try {
                    stack.push(item);
                } catch (const StackFullException& e) {
                    cout << e.what();
                }
                break;
            }
            case 2:
                try {
                    stack.pop();
                } catch (const StackEmptyException& e) {
                    cout << e.what();
                }
                break;
        }
        cout << endl;
    }
    return 0;
}
